File: app/src/main/java/bigbigdw/joaradw/main/Tab_ViewModel.java
package bigbigdw.joaradw.main;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.Transformations;
import androidx.lifecycle.ViewModel;

public class Tab_ViewModel extends ViewModel {

    private MutableLiveData<Integer> mIndex = new MutableLiveData<>();
    private LiveData<String> mText = Transformations.map(mIndex, input -> "TAB" + input);

    public void setIndex(int index) {
        mIndex.setValue(index);
    }

    public LiveData<String> getText() {
        return mText;
    }
}

File: app/src/main/java/bigbigdw/joaradw/novel/TabViewModel.kt
package bigbigdw.joaradw.novel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.LiveData
import androidx.lifecycle.Transformations

class TabViewModel : ViewModel() {
    private val mIndex = MutableLiveData<Int>()
    val text = Transformations.map(mIndex) { input: Int -> "TAB$input" }
    fun setIndex(index: Int) {
        mIndex.value = index
    }
}